src/drivers/uavcan_v1/Uavcan.cpp

```diff
@@ -139,65 +150,72 @@ int UavcanNode::start(uint32_t node_id, uint32_t bitrate)
 	return PX4_OK;
 }
 
+void UavcanNode::init()
+{
+	// interface init
+	if (_can_interface) {
+		if (_can_interface->init() == PX4_OK) {
+
+			// We can't accept just any message; we must fist subscribe to specific IDs
+
+			// Subscribe to messages uavcan.node.Heartbeat.
+			canardRxSubscribe(&_canard_instance,
+					  CanardTransferKindMessage,
+					  uavcan_node_Heartbeat_1_0_FIXED_PORT_ID_,
+					  uavcan_node_Heartbeat_1_0_EXTENT_BYTES_,
+					  CANARD_DEFAULT_TRANSFER_ID_TIMEOUT_USEC,
+					  &_heartbeat_subscription);
+
+			// Subscribe to messages uavcan.node.NodeIDAllocationData_1_0 for PNP V1
+			canardRxSubscribe(&_canard_instance,
+					  CanardTransferKindMessage,
+					  uavcan_pnp_NodeIDAllocationData_1_0_FIXED_PORT_ID_,
+					  uavcan_pnp_NodeIDAllocationData_1_0_EXTENT_BYTES_,
+					  CANARD_DEFAULT_TRANSFER_ID_TIMEOUT_USEC,
+					  &_pnp_v1_subscription);
+
+			canardRxSubscribe(&_canard_instance,
+					  CanardTransferKindResponse,
+					  uavcan_register_Access_1_0_FIXED_PORT_ID_,
+					  uavcan_register_Access_Response_1_0_EXTENT_BYTES_,
+					  CANARD_DEFAULT_TRANSFER_ID_TIMEOUT_USEC,
+					  &_register_access_subscription);
+
+			canardRxSubscribe(&_canard_instance,
+					  CanardTransferKindResponse,
+					  uavcan_register_List_1_0_FIXED_PORT_ID_,
+					  uavcan_register_List_Response_1_0_EXTENT_BYTES_,
+					  CANARD_DEFAULT_TRANSFER_ID_TIMEOUT_USEC,
+					  &_register_list_subscription);
+
+
+
+			canardRxSubscribe(&_canard_instance,
+					  CanardTransferKindMessage,
+					  bms_port_id,
+					  reg_drone_service_battery_Status_0_1_EXTENT_BYTES_,
+					  CANARD_DEFAULT_TRANSFER_ID_TIMEOUT_USEC,
+					  &_drone_srv_battery_subscription);
+
+			canardRxSubscribe(&_canard_instance, //Temporory GPS message DSDL not defined yet
```

Comment:
It is though, you are using `reg.drone.physics.kinematics.geodetic.Point.0.1` in an adjacent file

Reply:
Ah yeah, this was just left in from the initial implementation PR.

Reply:
I just realized that a GNSS service is not expected to publish Point or any of its subtypes. You probably meant PointStateVarTs. If you subscribe to Point, you will get mangled data because the first value is a timestamp.

Reply:
Yeah, sorry, I just was using the `Point` topic for simplicity of getting things up and running.  Will mark a TODO to change it.  Still need the DilutionOfPrecision and possibly other related topics.